### PR TITLE
Companies model DB and Validations

### PR DESCRIPTION
**Pull request by DenisRem**
**CHECK LIST**
- [x] Create model
- [x] Create enum field
- [x] code coverage >90%
- [x] DB indexes added if needed
- [x] seed.rb file is updated if needed

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -49,3 +49,6 @@ yarn-debug.log*
 *.swl
 *.swm
 *.swn
+
+/config/database.yml
```
Comment (mmarusyk) [inline .gitignore:53]: Delete this line

File: config/database.yml [hunk 2]
```diff
@@ -80,6 +80,6 @@ test:
 #
 production:
   <<: *default
-  database: OSBB_production
-  username: OSBB
-  password: <%= ENV['OSBB_DATABASE_PASSWORD'] %>
+  database: OSBB
+  username: moskvyn
+  password: <%= ENV['ihormoskvyn211998'] %>
```
Comment (mmarusyk) [inline config/database.yml:85]: Don''t use yourself values

File: .gitignore [hunk 3]
```diff
@@ -49,3 +49,6 @@ yarn-debug.log*
 *.swl
 *.swm
 *.swn
+
+config/database.yml
```
Comment (chu4) [inline .gitignore]: remove from here it should not be in ignore

File: app/models/company.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
+class Company < ApplicationRecord
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/.freeze
+    VALID_PHONE = /\A\d{10}\z/.freeze
+    enum COMPANY_TYPE: { water_supply: 0, heating: 1, rent_payment: 2, accommodation_payment: 3,
```
Comment (chu4) [inline app/models/company.rb]: where are constants for each type?

Comment (DenisRem) [reply app/models/company.rb]: WATER_SUPPLY = 1
  HEATING = 2
 RENT_PAYMENT = 3
ACCOMMODATION_PAYMENT = 4
GAS = 5
ELECTRICITY = 6
GARBAGE_REMOVAL = 7
INTERNET = 8
INTERCOM = 9
TV = 10
OTHER = 11
enum COMPANY_TYPE: { WATER_SUPPLY => 'water_supply', HEATING => 'heating', RENT_PAYMENT => 'rent_payment' ....... }

Comment (chu4) [reply app/models/company.rb]:
```
  INSURER = "insurer"
  ADJASTOR = "adjastor"

  ROLES = [
    INSURER,
    ADJASTOR
  ]

  enum role: ROLES

```

Comment (chu4) [reply app/models/company.rb]: an example

File: config/database.yml [hunk 5]
```diff
@@ -23,7 +23,7 @@ default: &default
 
 development:
   <<: *default
-  database: OSBB_development
+  database: OSBB
```
Comment (BogdanBusko) [inline config/database.yml:26]: its bad idea to set same name for all dbs

File: config/database.yml (hunk 2 above)
Comment (BogdanBusko) [inline config/database.yml]: discard this changes

File: spec/models/company_spec.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
+require 'rails_helper'
+
+RSpec.describe Company, type: :model do
+  let!(:company) { FactoryBot.build(:company, name: "IFBut", phone: 2476316898, 
+                                    email: "ifbut@gmail.com", fax: 9762578517) }
+
+  describe 'validations' do
```
Comment (BogdanBusko) [inline spec/models/company_spec.rb:9]: add this gem for model testing https://github.com/thoughtbot/shoulda-matchers

File: config/database.yml [hunk 7]
```diff
@@ -80,6 +80,6 @@ test:
 #
 production:
   <<: *default
-  database: OSBB_production
-  username: OSBB
-  password: <%= ENV['OSBB_DATABASE_PASSWORD'] %>
+  database: OSBB
+  username: osbb
+  password: <%= ENV['osbb'] %>
```
Comment (BogdanBusko) [inline config/database.yml:85]: DON'T PUSH THIS FILE

File: app/models/company.rb [hunk 8]
```diff
@@ -0,0 +1,25 @@
+class Company < ApplicationRecord
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/.freeze
```
Comment (BogdanBusko) [inline app/models/company.rb:2]: you can use rails email regexp URI::MailTo::EMAIL_REGEXP

File: app/models/company.rb [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+  OTHER = "other".freeze
+  COMPANY_TYPE = [WATER_SUPPLY, HEATING, RENT_PAYMENT, ACCOMMODATION_PAYMENT, GAS,
+                  ELECTRICITY, GARBAGE_REMOVAL, INTERNET, INTERCOM, TV,
+                  ELEVATOR, OTHER].freeze
+  enum company_type: COMPANY_TYPE
+  before_save { email.downcase! }
+
+  validates :name, presence: true, length: { maximum: 50 }
+  validates :phone, presence: true, format: { with: VALID_PHONE }
+  validates :email, presence: true, length: { maximum: 255 }, format: { with: VALID_EMAIL_REGEX }
```
Comment (BogdanBusko) [inline app/models/company.rb:24]:
```ruby
validates :email, presence: true, length: { maximum: 255 }, format: { with: URI::MailTo::EMAIL_REGEXP }
```

Comment (BogdanBusko) [reply app/models/company.rb:24]: and why email can be so long?:)

File: app/models/company.rb [hunk 10]
```diff
@@ -0,0 +1,28 @@
+class Company < ApplicationRecord
+  VALID_PHONE = /\A\d{10}\z/.freeze
```
Comment (chu4) [inline app/models/company.rb:2]: add frozen string literal at the top of the file and remove .freeze on constants

File: app/models/company.rb [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+  ACCOMMODATION_PAYMENT = "accommodation_payment".freeze
+  GAS = "gas".freeze
+  ELECTRICITY = "electricity".freeze
+  GARBAGE_REMOVAL = "garbage_removal".freeze
+  INTERNET = "internet".freeze
+  INTERCOM = "intercom".freeze
+  TV = "tv".freeze
+  ELEVATOR = "elevator".freeze
+  OTHER = "other".freeze
+  COMPANY_TYPE = [WATER_SUPPLY, HEATING, RENT_PAYMENT, ACCOMMODATION_PAYMENT, GAS,
```
Comment (chu4) [inline app/models/company.rb:15]: put new line after each type and after opening/ before closing brackets

File: config/database.yml [hunk 12]
```diff
@@ -19,14 +19,14 @@ default: &default
   encoding: unicode
   username: <%= ENV['OSBB_DATABASE_USERNAME'] %>
   password: <%= ENV['OSBB_DATABASE_PASSWORD'] %>
-  host: localhost
```
Comment (chu4) [inline config/database.yml:22]: pls stop editing this file at your own will
configuration is set for every one
communicate with a team of how to use common setup not your personal

File: test/test_helper.rb [hunk 13]
```diff
@@ -11,3 +11,10 @@ class ActiveSupport::TestCase
 
   # Add more helper methods to be used by all tests here...
 end
+
+Shoulda::Matchers.configure do |config|
```
Comment (chu4) [inline test/test_helper.rb:15]: this whole test directory should be removed we use rspec

File: app/models/company.rb [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+  enum company_types: COMPANY_TYPE
+  before_save { email.downcase! }
+
+  validates :name, presence: true
+  validates :name, length: { maximum: 50 }
+  validates :phone, presence: true
+  validates :phone, format: { with: VALID_PHONE }
+  validates :email, presence: true
+  validates :email, length: { maximum: 50 }
+  validates :email, format: { with: URI::MailTo::EMAIL_REGEXP }
```
Comment (chu4) [inline app/models/company.rb:27]: pls rebase and use annotate to add comments with DB fields for your model

File: test/test_helper.rb [hunk 15]
```diff
@@ -11,10 +11,3 @@ class ActiveSupport::TestCase
 
   # Add more helper methods to be used by all tests here...
 end
-
-Shoulda::Matchers.configure do |config|
```
Comment (BogdanBusko) [inline test/test_helper.rb]: Why you remove this code?

Comment (DenisRem) [reply test/test_helper.rb]: chu4 3 hours ago Member
this whole test directory should be removed we use rspec

File: app/models/company.rb [hunk 16]
```diff
@@ -0,0 +1,59 @@
+# frozen_string_literal: true
+
+class Company < ApplicationRecord
+  VALID_PHONE = /\A\d{10}\z/.freeze
```
Comment (chu4) [inline app/models/company.rb:4]: last .freeze left

**Review by chu4 — APPROVED** [review]

**Review by Andrii-Kozak — APPROVED** [review]

**Review by Ukhanskyi — APPROVED** [review]

**Pull request merged**